src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -221,21 +422,10 @@ public Delegate[] CreateGetters(IRow input, Func<int, bool> activeOutput, out Ac
 
             public RowMapperColumnInfo[] GetOutputColumns()
             {
-                return new[] { new RowMapperColumnInfo(_outputColName, _outputColType, null) };
-            }
-
-            private static (ColumnType, TFDataType) GetOutputTypes(TFGraph graph, string columnName)
-            {
-                Contracts.AssertValue(graph);
-                Contracts.AssertNonEmpty(columnName);
-                Contracts.AssertValue(graph[columnName]);
-
-                var tfoutput = new TFOutput(graph[columnName]);
-                var shape = graph.GetTensorShape(tfoutput);
-
-                int[] dims = shape.ToIntArray().Skip(shape[0] == -1 ? BatchSize : 0).ToArray();
-                var type = TensorFlowUtils.Tf2MlNetType(tfoutput.OutputType);
-                return (new VectorType(type, dims), tfoutput.OutputType);
+                var info = new RowMapperColumnInfo[_parent.Outputs.Length];
+                for (int i = 0; i < _parent.Outputs.Length; i++)
+                    info[i] = new RowMapperColumnInfo(_parent.Outputs[i], _parent.OutputTypes[i], null);
+                return info;
             }
 
             private static (string[], int[], bool[], TFShape[], TFDataType[]) GetInputMetaData(TFGraph graph, string[] source, ISchema inputSchema)
```

Comment:
not needed anymore

Reply:
are we doing the shape checks somewhere ? e.g. lines 451-158 

---
In reply to: [215430955](https://github.com/dotnet/machinelearning/pull/840#discussion_r215430955) [](ancestors = 215430955)

Reply:
They all got moved to TensorFlowMapper constructor.

---
In reply to: [215438936](https://github.com/dotnet/machinelearning/pull/840#discussion_r215438936) [](ancestors = 215438936,215430955)